File: Pythonleetcode/Leetcode7.py
def reverse( x):
    x=str(x)
    if x[0]=='-':
        a=x[1:]
        a=a[::-1]
        x='-'+a
        if int(x)>=-(2**31) and int(x)<=2**31-1:
            return int(x)
        return 0
    x=x[::-1]

    if int(x) >= -(2 ** 31) and int(x) <= 2 ** 31 - 1:
        return int(x)

    return 0

print(reverse(123))

File: CppPractice/Project1/Project1/Leetcode7.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

using namespace std;

class Solution {
public:
    int reverse(int x) {
        if (x>=INT_MAX or x<=INT_MIN)
        {
            return 0;
        }
        bool is_neg;
        string x_str;
        int temp;
        if (x<0)
        {
            temp = -x;
            is_neg = true;
        }
        else
        {
            temp = x;
            is_neg = false;
        }
        x_str = to_string(temp);
        ::reverse(x_str.begin(),x_str.end());
        if (is_neg)
        {
            x_str = "-" + x_str;

        }
        try {
            int ans = stoi(x_str);
            return ans;
        }
        catch (const std::out_of_range& e) {
            return 0;
        }

    }
};